### PR TITLE
Add troubleshooting section about node.js

### PR DESCRIPTION
**Pull request by rchiodo**
I'll update the aka.ms link in the pyrx PR to point to this section after this goes through.

### CODE REVIEW

**Review by bschnurr — APPROVED** [review]

File: TROUBLESHOOTING.md [hunk 1]
```diff
@@ -89,6 +89,18 @@ you are running an official build and are still running into issues, please file
 If you are using Pylance in a WSL environment, make sure your workspace is located under a WSL folder (`/home/...`) and not shared with Windows (`/mnt/...`).
 See issues [#1443](https://github.com/microsoft/pylance-release/issues/1443#issuecomment-867863124) and [vscode-remote-release#5000](https://github.com/microsoft/vscode-remote-release/issues/5000).
 
+### Pylance is crashing
+
+Although we attempt to prevent Pylance from crashing, sometimes certain configurations can cause problems for Pylance. One particular problem is the amount of memory that Pylance is allowed to allocate when running inside of VS Code. VS Code ships with [pointer compression](https://www.electronjs.org/blog/v8-memory-cage) enabled. This makes VS Code run faster, but limits the amount of memory that Pylance can use. With some configurations, we may need more than 4GB of memory in order to analyze your project. 
+
+If you think you're hitting an OOM situation, you can alleviate this problem by providing your own [Node.js](https://nodejs.org/en/download/) executable to run Pylance with. Pylance (by default) runs using VS code's Node.js executable (which has the 4GB limit).
```
Comment (luabud) [inline TROUBLESHOOTING.md:96]:
```suggestion
If you think you're hitting an OOM (out-of-memory) situation, you can alleviate this problem by providing your own [Node.js](https://nodejs.org/en/download/) executable to run Pylance with. Pylance (by default) runs using VS Code's Node.js executable (which has the 4GB limit). 
```

File: TROUBLESHOOTING.md [hunk 2]
```diff
@@ -89,6 +89,18 @@ you are running an official build and are still running into issues, please file
...
 See issues [#1443](https://github.com/microsoft/pylance-release/issues/1443#issuecomment-867863124) and [vscode-remote-release#5000](https://github.com/microsoft/vscode-remote-release/issues/5000).
 
+### Pylance is crashing
+
+Although we attempt to prevent Pylance from crashing, sometimes certain configurations can cause problems for Pylance. One particular problem is the amount of memory that Pylance is allowed to allocate when running inside of VS Code. VS Code ships with [pointer compression](https://www.electronjs.org/blog/v8-memory-cage) enabled. This makes VS Code run faster, but limits the amount of memory that Pylance can use. With some configurations, we may need more than 4GB of memory in order to analyze your project. 
+
+If you think you're hitting an OOM situation, you can alleviate this problem by providing your own [Node.js](https://nodejs.org/en/download/) executable to run Pylance with. Pylance (by default) runs using VS code's Node.js executable (which has the 4GB limit). 
+
+To specify your own Node.js executable, set this setting in your user settings.json and restart VS Code:
```
Comment (luabud) [inline TROUBLESHOOTING.md:98]:
```suggestion
To specify your own Node.js executable, set this setting in your User settings.json and restart VS Code:
```

**Review by luabud — APPROVED** [review]
thank you so much!

File: TROUBLESHOOTING.md (hunk 1 above)
Comment (debonte) [inline TROUBLESHOOTING.md:96]: I'd suggest expanding "OOM" to "out of memory", or alternatively adding "(OOM)" after the first mention of "out of memory" above.

**Review by debonte — APPROVED** [review]

File: TROUBLESHOOTING.md (hunk 1 above)
Comment (debonte) [reply TROUBLESHOOTING.md]: Maybe just replace "OOM" with "out-of-memory". I don't think providing the acronym buys us anything.

File: TROUBLESHOOTING.md [hunk 3]
```diff
@@ -89,6 +89,18 @@ you are running an official build and are still running into issues, please file
 If you are using Pylance in a WSL environment, make sure your workspace is located under a WSL folder (`/home/...`) and not shared with Windows (`/mnt/...`).
 See issues [#1443](https://github.com/microsoft/pylance-release/issues/1443#issuecomment-867863124) and [vscode-remote-release#5000](https://github.com/microsoft/vscode-remote-release/issues/5000).
 
+### Pylance is crashing
+
+Although we attempt to prevent Pylance from crashing, sometimes certain configurations can cause problems for Pylance. One particular problem is the amount of memory that Pylance is allowed to allocate when running inside of VS Code. VS Code ships with [pointer compression](https://www.electronjs.org/blog/v8-memory-cage) enabled. This makes VS Code run faster, but limits the amount of memory that Pylance can use. With some configurations, we may need more than 4GB of memory in order to analyze your project. 
+
+If you think you're hitting an OOM (out-of-memory) situation, you can alleviate this problem by providing your own [Node.js](https://nodejs.org/en/download/) executable to run Pylance with. Pylance (by default) runs using VS Code's Node.js executable (which has the 4GB limit).
```
Comment (debonte) [inline TROUBLESHOOTING.md:96]:
```suggestion
If you think you're hitting an out-of-memory situation, you can alleviate this problem by providing your own [Node.js](https://nodejs.org/en/download/) executable to run Pylance with. Pylance (by default) runs using VS Code's Node.js executable (which has the 4GB limit). 
```

**Review by StellaHuang95 — APPROVED** [review]

**Review by luabud — APPROVED** [review]

**Pull request merged**